apps/wallet-mobile/src/Staking/PoolTransition/usePoolTransition.tsx

```diff
@@ -0,0 +1,426 @@
+import {useSelectedWallet} from '../../features/WalletManager/Context'
+import {formatTimeSpan, Quantities} from '../../yoroi-wallets/utils'
+
+// TODO: PoolTransition and getPoolTransition will come from yoroi-lib
+type PoolTransition = {
+  currentPool: PoolInfo & {tax: number}
+  suggestedPool: PoolInfo & {tax: number}
+  deadlineMilliseconds: number
+}
+
+const getPoolTransition = async (currentPoolId: string): Promise<PoolTransition | null> => {
+  return new Promise((resolve) => {
+    if (currentPoolId === 'd248ded3c18e0e80d07a46f00a2d808075b989ccb1a0e40a76e5cee1') {
+      resolve({
+        currentPool: {
+          pic: 'https://img.cexplorer.io/a/5/5/f/2/pool12eht6dqxpzqj87xuextrpufz2gxmt4reuesuw26r2utzw0kw906.png',
+          ticker: 'EMRG',
+          name: 'emurgo old',
+          roa: '0',
+          tax: '3.5',
+        } as unknown as PoolInfo & {tax: number},
+        suggestedPool: {
+          pic: 'https://img.cexplorer.io/e/c/2/3/1/pool1dkww2vlysa8lsnuf5rca979zdsyr3zvt59hu7e420yxfunkka2z.png',
+          id: 'df1750df9b2df285fcfb50f4740657a18ee3af42727d410c37b86207',
+          name: 'emurgo new',
+          roa: '5.1',
+          ticker: 'EMRG',
+          tax: '3.2',
+        } as unknown as PoolInfo & {tax: number},
+        deadlineMilliseconds: 2999777000,
+      })
+    } else {
+      resolve(null)
+    }
+  })
+}
+
+export const usePoolTransition = () => {
+  const navigation = useNavigation()
+  const wallet = useSelectedWallet()
+  const {stakingInfo} = useStakingInfo(wallet)
+
+  const isStaked = stakingInfo?.status === 'staked'
+  const currentPoolId = isStaked ? stakingInfo?.poolId : ''
+
+  const poolTransitionQuery = useQuery({
+    enabled: isStaked,
+    retry: false,
+    staleTime: 5 * 60 * 1000,
+    queryKey: [wallet.id, 'poolTransition', currentPoolId],
+    queryFn: () => getPoolTransition(currentPoolId),
+  })
+
+  const poolTransition = poolTransitionQuery.data ?? null
+  const poolId = poolTransition?.suggestedPool.id ?? ''
+
+  const stakingTxQuery = useQuery({
+    enabled: poolId !== '',
+    retry: false,
+    staleTime: 5 * 60 * 1000,
+    queryKey: [wallet.id, 'stakingTx', poolId],
+    queryFn: async () => {
+      const accountStates = await wallet.fetchAccountState()
+      const accountState = accountStates[wallet.rewardAddressHex]
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+      if (!accountState) throw new Error('Account state not found')
+
+      const stakingUtxos = await wallet.getAllUtxosForKey()
+      const amountToDelegate = Quantities.sum([
+        ...stakingUtxos.map((utxo) => utxo.amount as Balance.Quantity),
+        accountState.remainingAmount as Balance.Quantity,
+      ])
+
+      return wallet.createDelegationTx(poolId, new BigNumber(amountToDelegate))
+    },
+  })
+
+  const yoroiUnsignedTx = stakingTxQuery.data
+
+  const navigateToUpdate = () =>
+    !yoroiUnsignedTx
```

Comment:
this is a UI responsability to disable it, the api should navigate.

Reply:
Navigate where? If the unsignedTx failed you can't navigate there. The UI is already checking canNavigate, which make sure this condition doesn't happen. 

We're discussing when to create the tx in other threads anyway, so resolving this one.

Reply:
yap, my point is the API, if a function is navigateToUpdate, when it's called it should navigate, it should be pure, it is relying on external context to decided whether is can/or/cannot navigate, this is the caller responsibility to check before calling it